restapi/src/main/java/io/stargate/web/docsapi/service/query/search/resolver/BaseResolver.java

```diff
@@ -61,6 +62,8 @@ public static DocumentsResolver resolve(
     // execute cnf optimization
     Expression<FilterExpression> cnf = RuleSet.toCNF(expression);
 
+    cnf = NegateFilters.resolve(cnf);
```

Comment:
I would add this rule to the expression parsing and immediately simplify there. We could potentially even eliminate some of the "stupid" conditions like `eq: x | not(neq: x)`. WDYT? 

I guess there is no need to take not out of the parsing..

Reply:
Please see my comment on `NegateFilters` (above) regarding simplification.

Regarding the `eq: x | not(neq: x)` example, the conversion to CNF includes simplification rules, so that should work already. I'll see if I can add a test case for that.

Reply:
Refactored the conversion code to support simplification after NOTs are eliminated (6db3bf767cc5a7918170c99be9ed2c1dddef9ba7)